cli/ast.rs

```diff
@@ -0,0 +1,654 @@
+          }
+        }
+        swc_ecmascript::ast::ExportSpecifier::Namespace(namespace_spec) => {
+          self.names.push(namespace_spec.name.sym.to_string());
+        }
+        // Default is only proposed syntax, not current supported, so noop
+        _ => {}
+      }
+    }
+  }
+
+  fn visit_export_default_decl(
+    &mut self,
+    node: &swc_ecmascript::ast::ExportDefaultDecl,
+    _parent: &dyn visit::Node,
+  ) {
+    match &node.decl {
+      swc_ecmascript::ast::DefaultDecl::Class(_) => {
+        self.names.push("default".to_string())
+      }
+      swc_ecmascript::ast::DefaultDecl::Fn(_) => {
+        self.names.push("default".to_string())
+      }
+      // Interface is a noop
+      _ => {}
+    }
+  }
+
+  fn visit_export_default_expr(
+    &mut self,
+    _node: &swc_ecmascript::ast::ExportDefaultExpr,
+    _parent: &dyn visit::Node,
+  ) {
+    self.names.push("default".to_string());
+  }
+
+  fn visit_export_all(
+    &mut self,
+    node: &swc_ecmascript::ast::ExportAll,
+    _parent: &dyn visit::Node,
+  ) {
+    self.export_all_specifiers.push(node.src.value.to_string());
+  }
+}
+
+/// Options which can be adjusted when transpiling a module.
+#[derive(Debug, Clone)]
+pub struct TranspileOptions {
+  /// When emitting a legacy decorator, also emit experimental decorator meta
+  /// data.  Defaults to `false`.
+  pub emit_metadata: bool,
+  /// Should the source map be inlined in the emitted code file, or provided
+  /// as a separate file.  Defaults to `true`.
+  pub inline_source_map: bool,
+  /// When transforming JSX, what value should be used for the JSX factory.
+  /// Defaults to `React.createElement`.
+  pub jsx_factory: String,
+  /// When transforming JSX, what value should be used for the JSX fragment
+  /// factory.  Defaults to `React.Fragment`.
+  pub jsx_fragment_factory: String,
+  /// Should JSX be transformed or preserved.  Defaults to `true`.
+  pub transform_jsx: bool,
+}
+
+impl Default for TranspileOptions {
+  fn default() -> Self {
+    TranspileOptions {
+      emit_metadata: false,
+      inline_source_map: true,
+      jsx_factory: "React.createElement".into(),
+      jsx_fragment_factory: "React.Fragment".into(),
+      transform_jsx: true,
+    }
+  }
+}
+
+/// A logical structure to hold the value of a parsed module for further
+/// processing.
+pub struct ParsedModule {
+  comments: SingleThreadedComments,
+  pub leading_comments: Vec<Comment>,
+  module: Module,
+  pub source_map: Rc<SourceMap>,
+}
+
+impl ParsedModule {
+  pub fn analyze_dependencies(&self) -> Vec<DependencyDescriptor> {
+    analyze_dependencies(&self.module, &self.source_map, &self.comments)
+  }
+
+  #[allow(dead_code)] // TODO(kitsonk) for bundling rewrite
+  pub fn analyze_exported_names(&self) -> (Vec<String>, Vec<String>) {
+    let mut collector = ExportCollector::default();
+    collector.visit_module(&self.module, &self.module);
+
+    (collector.names, collector.export_all_specifiers)
+  }
```

Comment:
Will it be needed with #7461?

Reply:
I thought we were going to try to leave the compiler alone until I got further with this refactor?  #7461 is going to be really disruptive.  I can drop this bit for now, but I really think we should hold off on attempting #7461.

Reply:
I think #7461 is a bit tangential to the whole compiler refactor - if we can get it working and passing current tests we should be alright? That'd be one less "request" type to TSC that we'd need to handle. Anyway, it won't happen anytime soon, just a suggestion. I'm happy to land this PR with this change 👍

Reply:
Most of it is in `tsc.rs` which is the compiler that is being refactored... 😕